libmultilabel/nn/plt.py

```diff
@@ -0,0 +1,481 @@
+            else:
+                selected = set()
+                for y in ys:
+                    y = y.item()
+                    if y in pos:
+                        selected.add(y)
+                    if len(selected) == self.predict_top_k:
+                        break
+                if len(selected) < self.predict_top_k:
+                    selected = (list(selected) + list(pos - selected))[: self.predict_top_k]
+            clusters_selected[i] = np.asarray(list(selected))
+
+        trainer = init_trainer(
+            self.checkpoint_dir,
+            epochs=self.epochs,
+            patience=self.patience,
+            early_stopping_metric=self.val_metric,
+            val_metric=self.val_metric,
+            silent=self.silent,
+            use_cpu=self.use_cpu,
+            limit_train_batches=self.limit_train_batches,
+            limit_val_batches=self.limit_val_batches,
+            limit_test_batches=self.limit_test_batches,
+            save_checkpoints=True,
+        )
+        trainer.checkpoint_callback.filename = f"{self.CHECKPOINT_NAME}1"
+
+        # train & valid dataloaders for training
+        train_dataloader = self.dataloader(
+            PLTDataset(
+                train_x,
+                train_y,
+                num_classes=self.num_classes,
+                mapping=clusters,
+                clusters_selected=clusters_selected,
+            ),
+            shuffle=self.shuffle,
+        )
+        val_dataloader = self.dataloader(
+            PLTDataset(
+                val_x,
+                val_y,
+                num_classes=self.num_classes,
+                mapping=clusters,
+                clusters_selected=val_clusters_pred,
+                cluster_scores=val_scores_pred,
+            ),
+        )
+
+        try:
+            network = getattr(networks, "AttentionXML_1")(
+                embed_vecs=self.embed_vecs, num_classes=len(self.classes), **dict(self.network_config)
+            )
+        except Exception:
+            raise AttributeError("Failed to initialize AttentionXML")
+
+        model_1 = PLTModel(
+            classes=self.classes,
+            word_dict=self.word_dict,
+            embed_vecs=self.embed_vecs,
+            network=network,
+            log_path=self.log_path,
+            learning_rate=self.learning_rate,
+            optimizer=self.optimizer,
+            momentum=self.momentum,
+            weight_decay=self.weight_decay,
+            lr_scheduler=self.lr_scheduler,
+            scheduler_config=self.scheduler_config,
+            val_metric=self.val_metric,
+            metric_threshold=self.metric_threshold,
+            monitor_metrics=self.monitor_metrics,
+            multiclass=self.multiclass,
+            loss_function=self.loss_function,
+            silent=self.silent,
+            save_k_predictions=self.predict_top_k,
+        )
+        torch.nn.init.xavier_uniform_(model_1.network.attention.attention.weight)
```

Comment:
add a comment:

For weights not initialized by the level-0 model, use uniform initialization